Index: bind.spec
===================================================================
RCS file: /cvs/pkgs/rpms/bind/F-12/bind.spec,v
retrieving revision 1.326
diff -u -r1.326 bind.spec
--- bind.spec	21 Sep 2009 10:41:04 -0000	1.326
+++ bind.spec	5 Oct 2009 07:49:09 -0000
@@ -74,7 +74,6 @@
 Requires(post): grep, chkconfig
 Requires(pre):  shadow-utils
 Requires(preun):chkconfig
-Requires:       dnssec-conf
 Obsoletes:      bind-config < 30:9.3.2-34.fc6, caching-nameserver < 31:9.4.1-7.fc8
 Provides:       bind-config = 30:9.3.2-34.fc6, caching-nameserver = 31:9.4.1-7.fc8
 BuildRequires:  openssl-devel, libtool, autoconf, pkgconfig, libcap-devel
@@ -100,6 +99,7 @@
 %package sdb
 Summary: BIND server with database backends and DLZ support
 Group:   System Environment/Daemons
 
 %description sdb
 BIND (Berkeley Internet Name Domain) is an implementation of the DNS
@@ -250,6 +250,7 @@
   --with-dlz-ldap=yes \
   --with-dlz-postgres=yes \
   --with-dlz-mysql=yes \
+  --with-dlz-bdb=yes \
   --with-dlz-filesystem=yes \
 %endif
 %if %{GSSTSIG}
