--- contrib/dlz/drivers/dlz_bdbhpt_driver.c.orig	2005-09-05 02:18:19.000000000 +0200
+++ contrib/dlz/drivers/dlz_bdbhpt_driver.c	2010-02-23 13:35:15.000000000 +0100
@@ -276,7 +276,22 @@
 
 	switch(db->client->get(db->client, NULL, &key, &data, DB_GET_BOTH)) {
 	case DB_NOTFOUND:
-		result = ISC_R_NOTFOUND;
+		/* not found, try with wildcard as well */
+		if (key.data != NULL)
+			free(key.data);
+		key.data = strdup("*");
+		key.size = 1;
+
+		switch(db->client->get(db->client, NULL, &key, &data, DB_GET_BOTH)){
+			case DB_NOTFOUND:
+				result = ISC_R_NOTFOUND;
+				break;
+			case 0:
+				result = ISC_R_SUCCESS;
+				break;
+			default:
+				result = ISC_R_FAILURE;
+		}
 		break;
 	case 0:
 		result = ISC_R_SUCCESS;
